executor/src/witgen/data_structures/identity.rs

```diff
@@ -267,6 +256,49 @@ fn convert_identity<T: FieldElement>(
     }
 }
 
+fn convert_phantom_bus_interaction<T: FieldElement>(
+    bus_interaction: &PhantomBusInteractionIdentity<T>,
+) -> Identity<T> {
+    // Detect receives by having a unary minus in the multiplicity
+    // TODO: We should instead analyze the range constraints of the
+    // multiplicity expression.
+    let (is_receive, multiplicity) = match &bus_interaction.multiplicity {
+        AlgebraicExpression::UnaryOperation(op) => match op.op {
+            AlgebraicUnaryOperator::Minus => (true, (*op.expr).clone()),
+        },
+        _ => (false, bus_interaction.multiplicity.clone()),
+    };
+    // By convention, we assume that the first tuple entry is the interaction ID.
+    // TODO: Instead, we should have a separate field in the phantom bus interaction type.
+    let interaction_id = match bus_interaction.tuple.0[0] {
+        AlgebraicExpression::Number(id) => id,
+        // TODO: Relax this for sends when implementing dynamic sends
+        _ => panic!("Expected first tuple entry to be a static ID"),
+    };
+    // Remove the interaction ID from the list of expressions.
+    let expressions = bus_interaction.tuple.0.iter().skip(1).cloned().collect();
+    let selected_tuple = SelectedExpressions {
+        selector: bus_interaction.latch.clone(),
+        expressions,
+    };
+    match is_receive {
```

Comment:
Doesn't clippy tell you to use `if is_receive { ... } else { ... }` instead?

Reply:
no